File: server/src/Models/DMToolkitModels/Instances/BackgroundInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class BackgroundInstance : IDefinitionInstance<BackgroundDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string AbilityScoreDefinitionPlusTwoId { get; set; } = string.Empty;
    public string AbilityScoreDefinitionPlusOneId { get; set; } = string.Empty;
    public string FeatInstanceId { get; set; } = string.Empty;
    public FeatInstance? FeatInstance { get; set; } = null;
    public string DefinitionId { get; set; } = string.Empty;
    public BackgroundDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/backgroundInstance.ts
import type { AbilityScoreDefinition } from "../definitions/abilityScoreDefinition";
import type { BackgroundDefinition } from "../definitions/backgroundDefinition";
import type { FeatInstance, FeatInstanceDto } from "./featInstance";

export interface BackgroundInstance {
    id: string;
    abilityScoreDefinitionPlusTwo: AbilityScoreDefinition | null;
    abilityScoreDefinitionPlusOne: AbilityScoreDefinition | null;
    featInstance: FeatInstance | null;
    selectedItemSet: boolean;
    definition: BackgroundDefinition;
}

export interface BackgroundInstanceDto {
    id: string;
    abilityScoreDefinitionPlusTwoId: string;
    abilityScoreDefinitionPlusOneId: string;
    featInstance: FeatInstanceDto;
    selectedItemSet: boolean;
    definitionId: string;
}